File: TapstreamPlugin.m
//
//  TapstreamPlugin.m
//  
//  Created by Stephan Bönnemann on 7/1/12.
//  Copyright (c) 2012 Stephan Bönnemann, All rights reserved.
//

#import "TapstreamPlugin.h"
#import "Tapstream/Tapstream.h"

@implementation TapstreamPlugin
	
- (void) fireEvent:(NSMutableArray*)arguments withDict:(NSMutableDictionary*)options
{
  
  NSString* eventName = [options valueForKey:@"eventName"];
  bool oneTimeOnly = [[options valueForKey:@"oneTimeOnly"] boolValue];
  
  if (oneTimeOnly) {
    [[Tapstream shared] fireEvent:eventName oneTimeOnly:YES];
  } else {
    [[Tapstream shared] fireEvent:eventName];
  }
  
  //NSLog(@"The fired Tapstream Event = %@", eventName);
    
}

@end

File: TapstreamPlugin.js
/**
 * Cordova TapStream Plugin
 * Copyright (c) 2012 Stephan Bönnemann
 *
 */
 
(function() {
  
var cordovaRef = window.PhoneGap || window.Cordova || window.cordova; // old to new fallbacks

function TapstreamPlugin() {};

TapstreamPlugin.prototype.fireEvent = function(eventName /* String */, oneTimeOnly /* Boolean*/) {
  var options = {
    eventName: eventName,
    oneTimeOnly: Boolean(oneTimeOnly)
  };
  return cordovaRef.exec("TapstreamPlugin.fireEvent",options);
};

cordovaRef.addConstructor(function() {
  if(!window.plugins) {
    window.plugins = {};
  }
  if (!window.plugins.tapstreamPlugin) {
    window.plugins.tapstreamPlugin = new TapstreamPlugin();
  }
});

})();
